pkg/metrics/exporter.go

```diff
@@ -49,19 +54,48 @@ func RegisterPrometheusExporter(registry *prom.Registry) (http.Handler, error) {
 // RegisterStackdriverExporter register a Stackdriver exporter to OpenCensus.
 // It will add Agones metrics into Stackdriver on Google Cloud.
 func RegisterStackdriverExporter(projectID string) (sd *stackdriver.Exporter, err error) {
-	// Default project will be used
+	instanceID, err := metadata.InstanceID()
+	if err != nil {
+		logger.WithError(err).Warn("error getting instance ID")
+		instanceID = unknown
+	}
+	zone, err := metadata.Zone()
+	if err != nil {
+		logger.WithError(err).Warn("error getting zone")
+		zone = unknown
+	}
+	clusterName, err := metadata.InstanceAttributeValue("cluster-name")
+	if err != nil {
+		logger.WithError(err).Warn("error getting cluster-name")
+		clusterName = unknown
+	}
+
 	sd, err = stackdriver.NewExporter(stackdriver.Options{
 		ProjectID: projectID,
 		// MetricPrefix helps uniquely identify your metrics.
 		MetricPrefix: "agones",
+		Resource: &monitoredres.MonitoredResource{
+			Type: "container",
```

Comment:
According to the Stackdriver document, I think "Type" should be "gke_container" or "k8s_container".
https://cloud.google.com/monitoring/custom-metrics/creating-metrics#monitored_resources_for_custom_metrics

(I tried to run this build then I got following error.
2019/11/06 12:11:41 Failed to export to Stackdriver: rpc error: code = InvalidArgument desc = One or more TimeSeries could not be written: Unrecognized resource type: container.: timeSeries[0-17])

Reply:
I used the driver documentation. But yeah I think you're right.

Reply:
changed, let me know.